src/core/geometry_utils.h

```diff
@@ -11,8 +11,35 @@
 #include <cmath>
 
 
-#define EQ_DOUBLE(a, b)                         \
-  (std::abs((a) - (b)) < 0.0001)
+inline bool equal(double a, double b) {
+  return (std::abs((a) - (b)) < 0.0001);
+}
+
+inline bool equal(double a, double b, double c) {
+  return (equal(a, b) && equal(a, c));
+}
+
+inline bool less_or_equal(double a, double b) {
+  return (equal(a, b) || (a < b));
+}
+
+inline bool is_between(double point, double left, double right) {
+  return (less_or_equal(left, point) && less_or_equal(point, right)) ||
+      (less_or_equal(right, point) && less_or_equal(point, left));
+}
+
+inline bool is_strictly_between(double point, double left, double right) {
+  return (((left < point) && (point < right)) ||
+        ((right < point) && (point < left)));
+}
+
+inline bool both_less(double x, double y, double threshold) {
```

Comment:
Will it be helpful for understanding?
if it is written in code for example
while(both_less(get_mean(), estimated_mean, prev_mean))...